typos.toml

```diff
@@ -0,0 +1,41 @@
+[files]
+extend-exclude = [
+    # ./compiler excludes
+    "compiler/rustc_codegen_gcc",
+    "compiler/rustc_codegen_cranelift",
+    "compiler/rustc_baked_icu_data",
+]
```

Comment:
Could you add a comment explaining why we exclude these?

Reply:
added